### DISCUSSION TITLE
Support for Gunicorn pre_request setting?

### DISCUSSION THREAD

**Discussion by gwerbin** [Kludex/uvicorn#2188] [Ideas] [answered]
Gunicorn has a [`pre_request` server hook](https://docs.gunicorn.org/en/stable/settings.html?highlight=logger#pre-request):

> Called just before a worker processes the request.

However in my testing, this function does not appear to be called when using Uvicorn workers, i.e. `worker_class = "uvicorn.workers.UvicornWorker"`.

Would it be difficult to add support for this server hook? It's useful as a light-duty alternative to ASGI middleware.

There is also a `post_request` server hook that would probably be convenient for some users.

Overall, it would at least be useful if the Uvicorn documentation stated which Gunicorn server hooks were/weren't supported by `UvicornWorker`.

**Comment by khaled5321** [comment]
Any update on this?

**Comment by Kludex** [comment, marked as answer]
- Let's follow this on https://github.com/Kludex/uvicorn-worker/issues/20
